src/main/java/org/stellar/base/PaymentOperation.java

```diff
@@ -57,7 +61,7 @@ public long getAmount() {
      * Construct a new PaymentOperation builder from a PaymentOp XDR.
      * @param op {@link PaymentOp}
      */
-    Builder(PaymentOp op) {
+    Builder(PaymentOp op) throws AssetCodeLengthInvalidException {
```

Comment:
It seems to me that AssetCodeLengthInvalidException should be a runtime exception instead of a checked exception.  In most instances, the code is going to be coming from external sources (either config files, user input, or the network) and having the compiler enforce the catching is going to be a pain.